### ISSUE TITLE
[Bug] Pause when not on wifi gets stuck

### ISSUE THREAD

**Issue by JohnADilley** [BillGoldsmith/iOS-bugfest#193]
## Brief description of the issue:

Hitting pause causes the iPhone app to get stuck, unable to play, when off wifi.

### Steps to reproduce:

Have a cellular connection but be out of range of wifi.  (Without cell access you get a different error.)
Play from a cache block.  In my case I was playing over headphones.
Hit pause icon on the app, or from the headphone.
Hit play to restart playback.

### Expected result:

Play should resume.

### Actual result:

Modal dialog box
    Warning
    Please note playback is blocked because you have 'stream on wifi only' set in preferences and you appear to be on a cellular network
    "Turn 'Wifi Only' off" or "OK" buttons

Pressing OK returns to the screen, play button still does not work.

Switching cache blocks will fix, causing playback from the playpoint of that block.
Switching back to the previous block starts playing from that block as intended.

### Where did you see it? (App build version, device type and iOS version):

RP Mk2, build 66, iPhone Xs running iOS 16.2

**Comment by RPNate1** [comment]
Thanks for plugging away at the DL manager. We need to sort out all of the cases as it's important to many users. We'll get this sorted. It's a complex beast.

Warnings should not appear constantly that interfere with UX.

**Comment by JohnADilley** [comment]
Update (Mk2 build 70)
- I tried to repro and was unable to get the hang to happen. I was off wifi but still on cellular. I was able to fast forward and switch channels. (I like the new "available channels" indicator in bottom-left selector).
- When switching channels and songs I _sometimes_ saw the warning dialog box. But it was not reliable and hasn't come back in a while.

If this is something you've addressed I can close it; just tag with Validate. If it changed mysteriously it's always good to understand why first :-).

NOTE: When I went offline (and I think I also restarted the app) I lost the pre-loaded block I had been listening to on the Main Mix. It would be nice for this preload not to be ditched.

DL Manager does seem like a beast! I worked in internet content caching for a significant part of my career. We were very busy for a lot of years so I feel for you!
